File: Product_APP_MVC/app/src/main/java/com/example/product_app_mvc/Network/ProductService.java
package com.example.product_app_mvc.Network;



import com.example.product_app_mvc.model.ProductsResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

// Interface for the Products API
public interface ProductService {
    // Get all products or apply a limit query if needed
    @GET("random.php")
    Call<ProductsResponse> getProducts();


    @POST("filter.php")
    Call<ProductsResponse> searchMealbyIngredient(@Query("i") String mealName);//search by Ingredient

    @POST("filter.php")
    Call<ProductsResponse> searchMealbyCountry(@Query("a") String CountryName);//search by Country

    @POST("filter.php")
    Call<ProductsResponse> searchMealbyCategory(@Query("c") String CountryName);//search by Category

}

File: Android_Development/05_Android_Kotlin/Myapp/MyApplication/app/src/main/java/com/example/myapplication/network/ProductService.kt
package com.example.myapplication.Network

import com.example.myapplication.model.Product
import com.example.myapplication.model.ProductsResponse
import kotlinx.coroutines.flow.Flow
import retrofit2.Response
import retrofit2.http.GET

interface ProductService {

    @GET("products")
    suspend fun getProducts(): ProductsResponse
}
